library/core/src/main/java/com/google/android/exoplayer2/text/ttml/TtmlNode.java

```diff
@@ -172,11 +181,37 @@ private void getEventTimes(TreeSet<Long> out, boolean descendsPNode) {
   }
 
   public List<Cue> getCues(long timeUs, Map<String, TtmlStyle> globalStyles,
-      Map<String, TtmlRegion> regionMap) {
+      Map<String, TtmlRegion> regionMap, Map<String, String> imageMap) {
+
     TreeMap<String, SpannableStringBuilder> regionOutputs = new TreeMap<>();
+    List<Pair<String, String>> regionImageList = new ArrayList<>();
+
     traverseForText(timeUs, false, regionId, regionOutputs);
     traverseForStyle(timeUs, globalStyles, regionOutputs);
+    traverseForImage(timeUs, regionId, regionImageList);
+
     List<Cue> cues = new ArrayList<>();
+
+    // Create image based cues
+    for (Pair<String, String> regionImagePair : regionImageList) {
+      String base64 = imageMap.get(regionImagePair.second);
+      byte[] decodedString = Base64.decode(base64, Base64.DEFAULT);
+      Bitmap decodedByte = BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
+      TtmlRegion region = regionMap.get(regionImagePair.first);
+
+      cues.add(
+          new Cue(decodedByte,
+              region.position,
+              Cue.ANCHOR_TYPE_MIDDLE,
```

Comment:
Is this definitely correct? As opposed to TYPE_UNSET, which is what's used for the position anchor in the text case below.

Reply:
Right, the `SubtitlePainter` should decide how to render this.

Reply:
I still think you should modify `setupBitmapLayout` to do something sensible if things are unset, and output it as unset from the decoder. There aren't any subtitle decoders setting unset anchors at the moment, so adding logic for that case in `setupBitmapLayout` should be fine (this is also probably why such logic is missing ;)).